source/simulator/solver_schemes.cc

```diff
@@ -992,6 +1033,10 @@ namespace aspect
         if ((particle_world.get() != nullptr) && (nonlinear_iteration > 0))
           particle_world->restore_particles();
 
+        // Apply a particle update if required by the particle properties
+        if (timestep_number > 0)
+          signals.post_restore_particles(*particle_world.get());
```

Comment:
Yes, I'll move it into the if condition.

I didn't trigger the signal in timestep 0, because the operator splitting on fields doesn't do so either. And for the specific stress update that I want to do, there is no update to do of the previous timestep during timestep 0. I wrote this in the comment on the same call to the signal but in a different solver scheme. Anyway, I'll remove the requirement.